pkg/block/blocktest/adapter.go

```diff
@@ -233,11 +243,47 @@ func testAdapterMultipartUpload(t *testing.T, adapter block.Adapter, storageName
 				multiParts[i].PartNumber = partNumber
 				multiParts[i].ETag = partResp.ETag
 			}
+
+			// List parts after upload
+			listResp, err := adapter.ListParts(ctx, obj, resp.UploadID, block.ListPartsOpts{})
+			if blockstoreType != block.BlockstoreTypeS3 {
+				require.ErrorIs(t, err, block.ErrOperationNotSupported)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, len(parts), len(listResp.Parts))
+				for i, part := range listResp.Parts {
+					require.Equal(t, multiParts[i].PartNumber, part.PartNumber)
+					require.Equal(t, int64(len(parts[i])), part.Size)
+					require.Equal(t, multiParts[i].ETag, part.ETag)
+					require.False(t, listResp.IsTruncated)
```

Comment:
I always prefer either to compare the entire struct for equality or to use `assert.Equal`.  That ensures that when the test fails I get as much information as possible.